### PR TITLE
docs(spec): Adds new Keyring protocol spec

### PR DESCRIPTION
**Pull request by thomastaylor312**
This PR proposes a new specification section for a "Keyring protocol."
The purpose of this protocol is to enable easier key distribution for the
80% use case. Instead of being something intrinsic to the bindle protocol,
I believe having this as a separate protocol that any server (including existing
key servers) could implement will make it much more flexible.

If this spec change is accepted, I will implement it in a followup PR

### CODE REVIEW

File: docs/keyring-protocol-spec.md
Comment (thomastaylor312) [inline docs/keyring-protocol-spec.md:25]: Please note that this is being kept simple on purpose for now. The types of parameters and filtering could be fairly complex in the future

**Review by vdice — COMMENTED** [review]
I wonder if we could use more doc additions/examples illustrating the setup of keys for bindle clients and servers.  For example, I'm a bindle client and I know the url to a bindle server and can authn with it.  How do I give the bindle server my public key(s) and how do I get the server public key(s)?  When the server and client are running on different hosts, neither has access to each other's keyrings, etc.

Oh, also, we should add the flow around signing invoices (`bindle sign-invoice`) and pushing bindles w/ signed invoices to the server.  Though perhaps this functionality has been around for some time... in that case, a separate docs issue would be appropriate.

**Comment by thomastaylor312** [comment]
Yeah I think the scope of this spec change is purely:

> how do I get the server public key(s)

Other key distribution methods are specifically excluded from this for now, but would definitely be a welcome addition.

The other suggestion about docs is definitely valid, but not in scope for this PR IMO. We should open an issue and update the docs in another PR

**Review by technosophos — APPROVED** [review]

**Comment by thomastaylor312** [comment]
@radu-matei and/or @fibonacci1729 could I get another review on this please? Since this is a spec change, I want to make sure at least one more maintainer reviews this

**Review by fibonacci1729 — APPROVED** [review]

**Pull request merged**
